test/webhook-apicoverage/tools/tools.go

```diff
@@ -186,3 +191,56 @@ func GetAndWriteTotalCoverage(webhookIP string, outputFile string) error {
 
 	return nil
 }
+
+// GetResourcePercentages calls resource percentage coverage API to retrieve
+// percentage values.
+func GetResourcePercentages(webhookIP string) (
+	*coveragecalculator.CoveragePercentages, error) {
+	client := &http.Client{Transport: &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	},
+	}
+
+	resp, err := client.Get(fmt.Sprintf(WebhookResourcePercentageCoverageEndPoint,
+		webhookIP))
+	if err != nil {
+		return nil, fmt.Errorf(
+			"encountered error making resource percentage coverage request:" +
+				" %v", err)
+	} else if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("invalid HTTP Status received for resource" +
+			" percentage coverage request. Status: %d", resp.StatusCode)
+	}
+
+	var body []byte
+	if body, err = ioutil.ReadAll(resp.Body); err != nil {
+		return nil, fmt.Errorf("error reading resource percentage" +
+			" coverage response: %v", err)
+	}
+
+	var coveragePercentages coveragecalculator.CoveragePercentages
+	if err = json.Unmarshal(body, &coveragePercentages); err != nil {
+		return nil, fmt.Errorf("error unmarshalling response to" +
+			" CoveragePercentages instance: %v", err)
+	}
+
+	return &coveragePercentages, nil
+}
+
+// WriteResourcePercentages writes CoveragePercentages to junit_xml output file.
+func WriteResourcePercentages(outputFile string,
+	coveragePercentages *coveragecalculator.CoveragePercentages) error {
+	if htmlData, err :=
+			view.GetCoveragePercentageXMLDisplay(coveragePercentages); err != nil {
+		return fmt.Errorf("error building coverage percentage xml file." +
+			" error: %v", err)
+	} else {
```

Comment:
Same here.
Else if not required and so you can unindent the `if`.
And in the end, just `return ioutil.WriteFile(...)`.

Reply:
Fixed